fri/src/verifier/mod.rs

```diff
@@ -313,7 +313,7 @@ where
             return Err(VerifierError::RemainderDegreeMismatch(max_degree_plus_1 - 1));
         }
         let offset: E::BaseField = self.options().domain_offset();
-
+        let remainder_poly: Vec<_> = remainder_poly.iter().copied().rev().collect();
```

Comment:
Let's add a comment here about why reversal is needed.

Reply:
I think the best approach would be to have say a `eval_horner_rev` function which would be just like `eval_horner` but without the `rev()`. This way we can remove the above line and it would be consistent with the goal of this PR, namely simplify/reduce the work on the verifier side.
What do you think?

Reply:
Sounds good!